src/App.tsx

```diff
@@ -15,36 +15,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_FIELD_ALPHABET = 'alphabet',
+  SORT_FIELD_LENGTH = 'length',
+  SORT_FIELD_RESET = '',
+}
+
+function getSortedArray(
+  array: string[],
+  sortField: string,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...array];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.SORT_FIELD_ALPHABET:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.SORT_FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [isReversed, setReverseField] = useState<boolean>(false);
+  const visibleGoods = getSortedArray(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={
+            sortField === SortType.SORT_FIELD_ALPHABET
+              ? `button is-info`
+              : `button is-info is-light`
+          }
+          onClick={() => {
+            setSortField(SortType.SORT_FIELD_ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={
+            sortField === SortType.SORT_FIELD_LENGTH
+              ? `button is-success`
+              : `button is-success is-light`
+          }
+          onClick={() => {
+            setSortField(SortType.SORT_FIELD_LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={
+            isReversed ? `button is-warning` : `button is-warning is-light`
+          }
+          onClick={() => setReverseField(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SortType.SORT_FIELD_RESET);
+              setReverseField(false);
+            }}
```

Comment:
Pls move the function inside `` handler to a separate const and use as a callback here